test/model/message_list_test.dart

```diff
@@ -271,6 +271,68 @@ void main() {
     checkInvariants(model);
   });
 
+  group('DeleteMessageEvent', () {
+    final stream = eg.stream();
+    final messages = List.generate(30, (i) => eg.streamMessage(stream: stream));
+
+    test('in narrow', () async {
+      await prepare(narrow: StreamNarrow(stream.streamId));
+      await prepareMessages(foundOldest: true, messages: messages);
+
+      check(model).messages.length.equals(30);
+      await store.handleEvent(eg.deleteMessageEvent(messages.sublist(0, 10)));
+      checkNotifiedOnce();
+      check(model).messages.length.equals(20);
+    });
+
+    test('not all in narrow', () async {
+      await prepare(narrow: StreamNarrow(stream.streamId));
+      await prepareMessages(foundOldest: true, messages: messages.sublist(5));
+
+      check(model).messages.length.equals(25);
+      await store.handleEvent(eg.deleteMessageEvent(messages.sublist(0, 10)));
+      checkNotifiedOnce();
+      check(model).messages.length.equals(20);
+    });
+
+    test('not in narrow', () async {
+      await prepare(narrow: StreamNarrow(stream.streamId));
+      await prepareMessages(foundOldest: true, messages: messages.sublist(5));
+
+      check(model).messages.length.equals(25);
+      await store.handleEvent(eg.deleteMessageEvent(messages.sublist(0, 5)));
+      checkNotNotified();
+      check(model).messages.length.equals(25);
+      checkInvariants(model);
+    });
+
+    test('complete message deletion', () async {
+      await prepare(narrow: StreamNarrow(stream.streamId));
+      await prepareMessages(foundOldest: true, messages: messages.sublist(0, 25));
+
+      check(model).messages.length.equals(25);
+      await store.handleEvent(eg.deleteMessageEvent(messages));
+      checkNotifiedOnce();
+      check(model).messages.length.equals(0);
+    });
+
+    test('non-consecutive message deletion', () async {
+      await prepare(narrow: StreamNarrow(stream.streamId));
+      await prepareMessages(foundOldest: true, messages: messages);
+      final messageToDelete = messages.sublist(2, 5) + messages.sublist(10, 15);
+
+      check(model).messages.length.equals(30);
+      await store.handleEvent(eg.deleteMessageEvent(messageToDelete));
+      checkNotifiedOnce();
+      final remainingMessageIds = [
+        ...messages.sublist(0, 2),
+        ...messages.sublist(5, 10),
+        ...messages.sublist(15)].map((message) => message.id).toSet();
+      check(model).messages.length.equals(22);
```

Comment:
nit:
```suggestion
      check(model).messages.length.equals(remainingMessageIds.length);
```

to make it less of a magic number